faststream/rabbit/broker/broker.py

```diff
@@ -655,3 +655,13 @@ async def declare_exchange(
         """Declares exchange object in **RabbitMQ**."""
         assert self.declarer, NOT_CONNECTED_YET  # nosec B101
         return await self.declarer.declare_exchange(exchange)
+    
+    async def ping(self, timeout: int | None) -> bool:
+        await super().ping(timeout)
+        try:
```

Comment:
Also, I didn't see `timeout` logic here. You can set use it in native methods or `anyio.move_on_after` method